amxmodx/hashing.cpp

```diff
@@ -0,0 +1,1963 @@
+// Copyright (c) 2014 Stephan Brumme. All rights reserved.
```

Comment:
I've looked a bit more what changes you did.
I think if we are going to provide a third-party library, this needs to be used as it is, unless there are necessary changes. Here removing std::string dependency is fine, but you should change only that.

So, this would be likely more appropriate to:
* Keep the original files organizations
* Move them in `public/hashlib` or similar name

About the `std::string` changes, did you test the safety?
You might as well use `ke::AString`.